### ISSUE TITLE
Electron crashes after file change

### ISSUE THREAD

**Issue by masiulis** [wallabyjs/public#597]
Tests running in electron work fine on load but crash after changing any file or test. This setup works with phantomJs2. Error message:

```
Uncaught TypeError: Cannot read property 'cache' of undefined
at http://localhost:64278/__modules/1.js?1463571726680:1

Uncaught TypeError: Cannot read property 'cache' of undefined
at http://localhost:64278/__modules/10.js?1463571726774:1

// and this goes on
```

### Wallaby.js configuration file

```javascript
const babel = require('babel-core');
const wallabyWebpack = require('wallaby-webpack');
const wallabyPostprocessor = wallabyWebpack({
    module: {
        loaders: [
            { test: /\.html$/, exclude: /node_modules/, loader: 'null-loader' },
            { test: /\.jpe?g$|\.svg$|\.png$/, exclude: /node_modules/, loader: 'null-loader' },
            { test: /\.json$/, loader: 'json-loader' },
            { test: /\.css$|\.scss$|\.sass$/, loader: 'style!css?modules!sass' },
        ],
    },
    externals: {
        'react/lib/ExecutionEnvironment': true,
        'react/lib/ReactContext': true,
    },
});

module.exports = function(wallaby) {
    return {
        files: [
            { pattern: 'src/**/*.js', load: false },
            { pattern: 'src/**/*.scss', load: false },
            { pattern: 'mock-server/**/*.js', load: false },
            { pattern: '!src/**/*.spec.js', load: false },
            { pattern: 'static/**/*.svg', load: false },
        ],
        tests: [
            { pattern: 'src/**/*.spec.js', load: false },
        ],
        postprocessor: wallabyPostprocessor,
        bootstrap: function() {
            window.__moduleBundler.loadTests();
        },
        env: {
            kind: 'electron',
        },
        compilers: {
            '**/*.js': wallaby.compilers.babel({
                babel: babel,
                presets: [
                    'react',
                    'es2015',
                    'stage-0',
                ],
            }),
        },
    };
};
```

### Dependencies
"electron-prebuilt": "^1.1.0"
"wallaby-webpack": "^0.0.22",
wallaby.js "version": "1.0.230", (according to wallaby/package.json)
Node: v6.1.0
WebStorm v2016
OSX 10.11.4

The only unordinary thing in our stack is that we are using scss modules in tests, so we have to use node v6.1.0 when running wallaby (otherwise we would get node-sass error for running different node versions in wallaby and webpack)

We would gladly switch from PhantomJS to Electron, but can't as long as tests are crashing on every second change. @ArtemGovorov, am I making an obvious configuration mistake and do you need an example repository?

**Comment by ArtemGovorov** [comment]
Hm, it's an interesting issue.

> am I making an obvious configuration mistake and do you need an example repository?

Your config looks good to me, so I'll need a repo where I could reproduce the issue. Would it be possible to create one?

**Comment by masiulis** [comment]
Thanks, I'll try to reproduce this bug in a new repo later today.

**Comment by ArtemGovorov** [comment]
Thanks. Could you please also check something for me to eliminate some possible reasons? 
Once you get the error that you've reported:
```
Uncaught TypeError: Cannot read property 'cache' of undefined
at http://localhost:xxxxx/__modules/1.js?1463571726680:1
```
try opening this page in Chrome: `http://localhost:xxxxx/wallaby_sandbox0.html` (note that the `xxxxx` port number may change every time you restart wallaby) and check the Elements tab of the Dev Tools to see if the fifth script (with `src`) from top is `wallaby-webpack.js`
![screen shot 2016-05-18 at 10 33 15 pm](https://cloud.githubusercontent.com/assets/979966/15358697/fb7fc0b2-1d48-11e6-9af6-b72b9201b2a4.png)

**Comment by masiulis** [comment]
I have created an example [https://github.com/masiulis/wallaby-electron](https://github.com/masiulis/wallaby-electron),  after ~1-10 changes (for example adding spaces) in the index.spec.js the test should crash. 

It looks like the deeper the import tree, the higher chance of crashing. For example, when I import `A -> B -> C -> store -> devTools` tests do crash, but if I skip A and import B, they do not (or they do so rarely that I could not reproduce).

I have opened `http://localhost:54153/wallaby_sandbox0.html` and I do in fact see
<img width="1026" alt="screen shot 2016-05-19 at 00 44 05" src="https://cloud.githubusercontent.com/assets/5903616/15375978/030ac626-1d5b-11e6-8be2-0d737e0b5551.png">

**Comment by ArtemGovorov** [comment]
Thanks for the repo. I wasn't able to reproduce the original issue (`Uncaught TypeError: Cannot read property 'cache' of undefined`), but I have reproduced another one that I believe happens for the exact same reason - Electron sometimes skips requests to load certain scripts. I'm not entirely sure why it happens, but I think I have found a way/hack to fix it. The fix is published in the latest core v1.0.231. Please restart your IDE and start wallaby to get the update, then try it and let me know how it goes.

**Comment by masiulis** [comment]
Strangely, example repo no longer throws errors, but our main repo now sometimes throws:
```
Uncaught Error: Cannot find module '661'
at http://localhost:57277/wallaby-webpack.js?1463641674964:1
```

I also saw errors in tracer.js twice (extremely rare compared to the upper one), don't know if they are related to this:
```
Uncaught TypeError: Cannot set property '0' of undefined
at http://localhost:56003/__wallaby__/tracer.js:14

// another one was something about .subscribe(), didn't manage to copy it before wallaby restarted
```

This problem seems to exist only in larger projects with many inner connections.

**Comment by ArtemGovorov** [comment]
Yep, I saw the error
```
Uncaught Error: Cannot find module 'xxx'
```
in the example repo (not the originally reported error), but it was fixed to me (and I can no longer reproduce it unfortunately). I tried adding longer dependency chains, but still can't reproduce it. I'm also using WebStorm on OSX.

How frequently does it now happen in your repo? Does the error disappear when you keep editing the code? Would it be possible to change the example repo to reproduce it? 

Not quite sure about the tracer error either. Does it only happen with Electron runner or you have seen with Phantom runner as well?

**Comment by ArtemGovorov** [comment]
I'll also rollback my previous hack/fix internally, because it looks like it's only delaying/hiding the error and will try to find another solution.

**Comment by ArtemGovorov** [comment]
Ok, thanks a lot to your repo, the issue was reliably reproduced, located and fixed. To get the fix, update your wallaby core to v1.0.232.

**Comment by masiulis** [comment]
Awesome, it works, thank you so much!
